src/audio/module_adapter/module/iadk_modules.c

```diff
@@ -80,8 +83,26 @@ static int iadk_modules_init(struct processing_module *mod)
 	uint32_t instance_id = IPC4_INST_ID(mod->dev->ipc_config.id);
 	uint32_t log_handle = (uint32_t) mod->dev->drv->tctx;
 	/* Connect loadable module interfaces with module adapter entity. */
-	void *mod_adp = system_agent_start(md->module_entry_point, module_id,
-					   instance_id, 0, log_handle, &mod_cfg);
+	/* Check if native Zephyr lib is loaded */
+	struct sof_man_fw_desc *desc;
+
+	desc = lib_manager_get_library_module_desc(module_id);
+	struct sof_man_module *module_entry =
+		    (struct sof_man_module *)((char *)desc + SOF_MAN_MODULE_OFFSET(0));
+
+	is_loadable = module_entry->type.lib_type;
+	void *mod_adp;
+
+	/* Check if module is sof loadable */
+	if (is_loadable) {
+		mod_adp = native_system_agent_start(md->module_entry_point, module_id, instance_id,
+						    0, log_handle, &mod_cfg,
+						    (void *)&dev->drv->ops, (void *)dev->drv->uid);
```

Comment:
type-casting to `void *` of other non-constant pointers shouldn't be needed

Reply:
done